Resolved server side filtering bug in All Vulnerabilities table (CRASM-397)

hawkishpolicy opened the pull request:
- Vulnerabilities table now maintains filters when calling new page of results.
- Filtered results are now paginated correctly.

# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##
- Applied filters from the filter model change were not being passed to the Pagination Model state when loading a new page.
- On clicking the next page button, the pagination model's filters key was being set to empty and just returning all results as if it was a page refresh. 
- By adding a filters key/value pair to the setPaginationModel state inside the fetch Vulnerabilities function, the filters remain in place while a new page is loaded.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##
- Filtered results are maintained and properly paginated.
- Closes #399
- Closes CRASM-397 
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->


## 📷 Screenshots (if appropriate) ##
![Screenshot 2024-07-08 at 3 51 54 PM](https://github.com/cisagov/XFD/assets/123108455/6b13ac2e-5254-4847-8066-45806610964d)
![Screenshot 2024-07-08 at 3 52 02 PM](https://github.com/cisagov/XFD/assets/123108455/df92c19c-c2dc-4ea9-937f-813b3a5eba7b)
![Screenshot 2024-07-08 at 3 52 14 PM](https://github.com/cisagov/XFD/assets/123108455/11958c2c-7526-40c9-9bd0-83e3d8c6d44d)

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Create a release.

hawkishpolicy committed:
Resolved server side filtering bug

- Vulnerabilities table now maintains filters when calling new page of results.
- Filtered results are now paginated correctly.

hawkishpolicy committed:
Refactored code for consistency & understanding

- Used similar state and function logic as All Domains for easier understanding and consistency.
- Added new piece of state titled "filters" to store the filters for the vulnerabilities.
- Filter model change now updated that state then passes it to the API call.
- This leaves the initialFilter state isolated and unchanged.

nickviola reviewed (approved):
LGTM

ameliav reviewed (changes requested):
The filtering is working, but the vulnerabilities table is out of date. Please merge the latest changes from develop

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into 399-ui-vulnerabilities-pagination-not-working-with-filtering-crasm-397

hawkishpolicy committed:
Changed errant "h4" to "p" in VulnerabilityCard

- quick change for a missed 'p' tag in the VulnerabilityCard component
- See All Vunerabilites link now displays properly.

ameliav reviewed (approved):
LGTM!
